From 3bc1df7fa1904ba66b7f4b1ac1682c24d9485083 Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Sat, 22 Sep 2018 11:21:53 +0200
Subject: [PATCH] find-dbgsym-packages: Ignore all vDSOs

The name of the vDSO varies with architecture.
Ignore vDSOs under all known names, not only "linux-vdso.so.*".

Fixes:

    $ find-dbgsym-packages /bin/ls
    eu-unstrip: linux-gate.so.1: No such file or directory
    "eu-unstrip" unexpectedly returned exit value 1 at /usr/bin/find-dbgsym-packages line 162.

on i386.
---
 find-dbgsym-packages | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/find-dbgsym-packages b/find-dbgsym-packages
index b319ca8..3e2ac21 100755
--- a/find-dbgsym-packages
+++ b/find-dbgsym-packages
@@ -44,7 +44,7 @@ foreach my $arg (@ARGV) {
     foreach my $id (keys %build_ids) {
         my ($path, $name) = @{$build_ids{$id}};
 
-        next if $name =~ /^linux-vdso[.]so[.]/;
+        next if $name =~ /^linux-(gate|vdso\d*)[.]so[.]/;
 
         my @p = get_debs_from_id($id);
         if (scalar @p == 0) {
-- 
2.19.0

